### PR TITLE
refactored plugins into features

### LINKED ISSUE #505: make every tauri plugin a cargo feature

### PR DESCRIPTION
**Pull request by crapStone**
closes #505

### CODE REVIEW

File: src-tauri/Cargo.toml [hunk 1]
```diff
@@ -27,14 +27,15 @@ tauri-build = { version = "1.0.0-beta-rc.0" }
...
 cabr2_load_save = { path = "./cabr2_load_save/", features = ["beryllium", "cabr2"] }
 cabr2_logger = { path = "./cabr2_logger/" }
-cabr2_search = { path = "./cabr2_search/", features = [ "tauri_plugin", "gestis" ] }
+cabr2_search = { path = "./cabr2_search/", features = ["gestis"] }
 log = "0.4.14"
-tauri = { version = "1.0.0-beta-rc.4", features =["api-all"]}
+tauri = { version = "1.0.0-beta-rc.4", features =["api-all"], optional = true }
 
 [features]
 custom-protocol = [ "tauri/custom-protocol" ]
```
Comment (6543) [inline src-tauri/Cargo.toml:35]:
```suggestion
```

we should move it into tauri_app if still needed

File: src-tauri/Cargo.toml [hunk 2]
```diff
@@ -27,14 +27,15 @@ tauri-build = { version = "1.0.0-beta-rc.0" }
...
 portable = [ "cabr2_config/portable" ]
-default = [ "custom-protocol", "portable" ]
+tauri_app = [ "tauri", "custom-protocol", "cabr2_config/tauri_plugin", "cabr2_load_save/tauri_plugin", "cabr2_search/tauri_plugin" ]
```
Comment (6543) [inline src-tauri/Cargo.toml:37]:
```suggestion
tauri_app = [ "tauri", "tauri/custom-protocol", "cabr2_config/tauri_plugin", "cabr2_load_save/tauri_plugin", "cabr2_search/tauri_plugin" ]
```

File: src-tauri/cabr2_logger/src/lib.rs [hunk 3]
```diff
@@ -1,31 +1,15 @@
 #![allow(clippy::new_without_default)]
 
-mod cmd;
+pub mod plugin;
```
Comment (6543) [inline src-tauri/cabr2_logger/src/lib.rs:3]:
```suggestion
#[cfg(feature = "tauri_plugin")]
pub mod plugin;
```

File: src-tauri/cabr2_logger/src/plugin.rs [hunk 4]
```diff
@@ -1,9 +1,12 @@
 use serde_json::{to_string_pretty, Value};
+use tauri::{plugin::Plugin, InvokeMessage, Params, Window};
 
 use cabr2_types::logging::LogLevel;
 
+use crate::setup_logger;
+
 #[tauri::command]
-pub fn log(level: LogLevel, path: String, messages: Vec<Value>) -> Result<(), String> {
```
Comment (6543) [inline src-tauri/cabr2_logger/src/plugin.rs:6]: this should resolve in changes at frontend too but there is no diff for that

File: src-tauri/Cargo.toml (hunk 1 above)
Comment (crapStone) [reply src-tauri/Cargo.toml]: this is a feature needed by the tauri cli, I will rebase out your commit

**Comment by 6543** [comment]
blocked by #1824

**Review by 6543 — APPROVED** [review]

**Pull request merged**